### ISSUE TITLE
Querybuilder generates invalid edgeql for invalid insert shape

### ISSUE THREAD

**Issue by jaclarke** [edgedb/edgedb-js#306]
From issue: https://github.com/edgedb/edgedb/issues/3647

If function returning shape object is used instead of plain shape object in `e.insert`:
```
e.insert(e.Person, () => ({
  firstName: 'john',
  lastName: 'smith'
}))
```
querybuilder generates the following incorrect edgeql:
```
INSERT default::Person {
  id
}
```

Typescript does catch this mistake, but we should still check at runtime and throw specific error instead of generating incorrect edgeql.

**Comment by internalfx** [comment]
I was also trying to insert a blank Person with the following query, not sure if this is related. But it is the same error.

```
    await e.insert(e.Person, {}).run(client)

    InternalServerError: invalid reference to FROM-clause entry for table "Person~1"
```

**Comment by Sikarii** [comment]
> Typescript does catch this mistake, but we should still check at runtime and throw specific error instead of generating incorrect edgeql.

It does not seem to catch the error for me on Typescript, the following does not throw TS errors:
```ts
    return e.insert(e.User, () => ({
      name: "Hello",
    })).run(client);
```

Using edgedb 0.19.15, Typescript 4.6.2 and client is freshly generated.

**Comment by jaclarke** [comment]
Yep, that should be valid, but the querybuilder also mistakenly injects an `id` into the shape here as well.
